annotated_test.go

```diff
@@ -599,6 +603,50 @@ func TestAnnotate(t *testing.T) {
 		assert.Len(t, got.sa, 2)
 	})
 
+	t.Run("Provide variadic function named with no given params", func(t *testing.T) {
+		t.Parallel()
+
+		var got *sliceA
+		app := fxtest.New(t,
+			fx.Provide(
+				fx.Annotate(newSliceA, fx.ParamTags(`name:"a"`)),
+			),
+			fx.Populate(&got),
+		)
+		defer app.RequireStart().RequireStop()
+		require.NoError(t, app.Err())
+
+		assert.Len(t, got.sa, 0)
+	})
+
+	t.Run("Invoke variadic function with multiple params", func(t *testing.T) {
+		t.Parallel()
+
+		app := fxtest.New(t,
+			fx.Supply(
+				fx.Annotate(newB(newA())),
```

Comment:
Looks like this has no annotation applied - why is this `Annotate`d?

Reply:
+1. I think fx.Annotate will no-op if there are no annotations.